sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -975,6 +976,22 @@ def test_callable_metric():
     assert_array_almost_equal(dist1, dist2)
 
 
+def test_algo_auto_metrics():
+    X = rng.rand(12, 3)
+    Xcsr = csr_matrix(X)
+    Valid_Metrics = dict(dense = ['precomputed', 'sqeuclidean',
+                             'yule', 'cosine', 'correlation'],
+                         sparse=PAIRWISE_DISTANCE_FUNCTIONS.keys())
```

Comment:
You can do away with this and just check for `VALID_METRICS['brute']` and `VALID_METRICS_SPARSE['brute']`. I think the other options are checked already.